core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -641,7 +618,7 @@ private[spark] class TaskSetManager(
       logInfo("Ignoring task-finished event for " + info.id + " in stage " + taskSet.id +
         " because task " + index + " has already completed successfully")
     }
-    failedExecutors.remove(index)
+    blacklistTracker.foreach(_.removeFailureExecutors(Iterable(info.executorId)))
```

Comment:
blacklistTracker.foreach(_.updateFailureExecutors(info, reason)) or to move this function to higher level